### PR TITLE
chore: [M3-6986] - Update AGLB endpoint, factories, and mocks

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
- Updates AGLB endpoint, factories, and mocks

## Major Changes 🔄
- Updates to match the latest API spec
- `Entry Points` have been renamed to `Configurations`
- Entities are now scoped per AGLB
  - Meaning that you now get routes by doing `/aglb/:id/routes` instead of `/aglb/routes`

## How to test 🧪
- Make sure that these types and endpoints match the API spec

### CODE REVIEW

File: packages/api-v4/src/aglb/routes.ts [hunk 1]
```diff
@@ -4,58 +4,80 @@ import { BETA_API_ROOT } from 'src/constants';
...
   );
 
 /**
- * getRoute
+ * getLoadbalancerRoute
  *
  * Returns an Akamai Global Load Balancer route
  */
-export const getRoute = (id: number) =>
+export const getLoadbalancerRoute = (loadbalancId: number, routeId: number) =>
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/routes.ts:22]:
```suggestion
export const getLoadbalancerRoute = (loadbalancerId: number, routeId: number) =>
```

File: packages/api-v4/src/aglb/routes.ts [hunk 2]
```diff
@@ -4,58 +4,80 @@ import { BETA_API_ROOT } from 'src/constants';
...
   Request<Route>(
-    setURL(`${BETA_API_ROOT}/aglb/routes/${encodeURIComponent(id)}`),
+    setURL(
+      `${BETA_API_ROOT}/aglb/${encodeURIComponent(
+        loadbalancId
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/routes.ts:26]:
```suggestion
        loadbalancerId
```

File: packages/api-v4/src/aglb/routes.ts [hunk 3]
```diff
@@ -4,58 +4,80 @@ import { BETA_API_ROOT } from 'src/constants';
...
   Request<Route>(
-    setURL(`${BETA_API_ROOT}/aglb/routes`),
+    setURL(`${BETA_API_ROOT}/aglb/${loadbalancerId}/routes`),
     setData(data),
     setMethod('POST')
   );
 
 /**
- * updateRoute
+ * updateLoadbalanacerRoute
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/routes.ts:48]:
```suggestion
 * updateLoadbalancerRoute
```

File: packages/api-v4/src/aglb/routes.ts [hunk 4]
```diff
@@ -4,58 +4,80 @@ import { BETA_API_ROOT } from 'src/constants';
...
  *
  * Updates an Akamai Global Load Balancer route
  */
-export const updateRoute = (id: number, data: Partial<RoutePayload>) =>
+export const updateLoadbalanacerRoute = (
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/routes.ts:52]:
```suggestion
export const updateLoadbalancerRoute = (
```

File: packages/api-v4/src/aglb/service-targets.ts [hunk 5]
```diff
@@ -10,13 +10,21 @@ import { BETA_API_ROOT } from 'src/constants';
...
 
 /**
- * getServiceTargets
+ * getLoadbalancerServiceTargets
  *
  * Returns a paginated list of Akamai Global Load Balancer service targets
  */
-export const getServiceTargets = (params: Params, filter?: Filter) =>
+export const getLoadbalancerServiceTargets = (
+  loadbalancrId: number,
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/service-targets.ts:18]:
```suggestion
  loadbalancerId: number,
```

File: packages/api-v4/src/aglb/service-targets.ts [hunk 6]
```diff
@@ -10,13 +10,21 @@ import { BETA_API_ROOT } from 'src/constants';
...
+  params?: Params,
+  filter?: Filter
+) =>
   Request<ResourcePage<ServiceTarget>>(
-    setURL(`${BETA_API_ROOT}/aglb/service-targets`),
+    setURL(
+      `${BETA_API_ROOT}/aglb/${encodeURIComponent(
+        loadbalancrId
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/service-targets.ts:25]:
```suggestion
        loadbalancerId
```

**Review by mjac0bs — COMMENTED** [review]
Spellcheck ✅ 

while I give the rest of this a closer look.

File: packages/manager/src/queries/aglb/serviceTargets.ts [hunk 7]
```diff
@@ -7,12 +7,16 @@ import {
 } from '@linode/api-v4/lib/types';
 import { useQuery } from 'react-query';
 
-export const queryKey = 'entrypoint';
+export const queryKey = 'aglbs';
```
Comment (mjac0bs) [inline packages/manager/src/queries/aglb/serviceTargets.ts:10]:
```suggestion
export const queryKey = 'aglb';
```
Are we intentionally using plural here to stick to convention with the other query keys? Do we also need to update the `queryKey` in `src/queries/aglb/loadbalancers.ts`?

File: packages/manager/src/queries/aglb/serviceTargets.ts [hunk 8]
```diff
@@ -7,12 +7,16 @@ import {
...
-export const useServiceTargetsQuery = (params: Params, filter: Filter) => {
+export const useServiceTargetsQuery = (
+  loadbalancerId: number,
+  params: Params,
+  filter: Filter
+) => {
   return useQuery<ResourcePage<ServiceTarget>, APIError[]>(
-    [`${queryKey}-list`, 'paginated', params, filter],
-    () => getServiceTargets(params, filter),
+    [queryKey, 'aglb', loadbalancerId, 'service-targets', params, filter],
```
Comment (mjac0bs) [inline packages/manager/src/queries/aglb/serviceTargets.ts:18]: Based on this line, the answer to my first question above is probably 'yes', but to help my own confusion: why do we need both "aglbs" and "aglb" here? 

image

File: packages/api-v4/src/aglb/configurations.ts [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+ *
+ * Returns a paginated list of Akamai Global Load Balancer configurations
+ */
+export const getLoadbalancerConfigurations = (
+  loadbalancerId: number,
+  params?: Params,
+  filter?: Filter
+) =>
+  Request<ResourcePage<Configuration>>(
+    setURL(`${BETA_API_ROOT}/aglb/${loadbalancerId}/configurations`),
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/configurations.ts:23]:
```suggestion
    setURL(`${BETA_API_ROOT}/aglb/${encodeURIComponent(loadbalancerId)}/configurations`),
```

File: packages/api-v4/src/aglb/routes.ts [hunk 10]
```diff
@@ -1,61 +1,95 @@
...
 
 /**
- * getRoutes
+ * getLoadbalancerRoutes
  *
  * Returns a paginated list of Akamai Global Load Balancer routes
  */
-export const getRoutes = () =>
+export const getLoadbalancerRoutes = (
+  id: number,
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/routes.ts:18]:
```suggestion
  routeId: number,
```
Just for consistency?

File: packages/api-v4/src/aglb/routes.ts [hunk 11]
```diff
@@ -1,61 +1,95 @@
...
+  params?: Params,
+  filter?: Filter
+) =>
   Request<ResourcePage<Route>>(
-    setURL(`${BETA_API_ROOT}/aglb/routes`),
-    setMethod('GET')
+    setURL(`${BETA_API_ROOT}/aglb/${encodeURIComponent(id)}/routes`),
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/routes.ts:23]:
```suggestion
    setURL(`${BETA_API_ROOT}/aglb/${encodeURIComponent(routeId)}/routes`),
```

File: packages/api-v4/src/aglb/types.ts [hunk 12]
```diff
@@ -3,53 +3,79 @@ export interface Loadbalancer {
...
+    label: string;
+  }[];
 }
 
-interface LoadbalancerPayload {
+export interface CreateLoadbalancerPayload {
   label: string;
   regions: string[];
   tags?: string[];
+  configuration_ids?: number[];
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/types.ts:17]: I'm only seeing `configurations` and not `configurations_id` in the API spec for POST /v4beta/aglb example request body. Did I miss something?

File: packages/manager/src/queries/aglb/serviceTargets.ts (hunk 7 above)
Comment (bnussman-akamai) [reply packages/manager/src/queries/aglb/serviceTargets.ts:10]: Yes, using `aglbs` to follow patter of `volumes`, `linodes`, etc..

I will consolidate those query keys.

File: packages/manager/src/queries/aglb/serviceTargets.ts (hunk 8 above)
Comment (bnussman-akamai) [reply packages/manager/src/queries/aglb/serviceTargets.ts:18]: We use that prefix just to help with invalidation later on. `["aglbs",  1, ...]` would probably be fine too but being a bit more specific is generally recommended for purposes of invalidation.

File: packages/api-v4/src/aglb/types.ts (hunk 12 above)
Comment (bnussman-akamai) [reply packages/api-v4/src/aglb/types.ts:17]: They must have removed that option! Will update types to reflect this!

File: packages/api-v4/src/aglb/types.ts [hunk 13]
```diff
@@ -3,53 +3,78 @@ export interface Loadbalancer {
...
+    service_targets: {
+      id: number;
+      label: string;
+      percentage: number;
+    }[];
+  }[];
 }
 
-export interface EntrypointPayload {
+export interface CreateRoutePayload {
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/types.ts:56]: Just noticed a list item in the API spec for POST /v4beta/aglb/{id}/routes
that says "The response body for routes will ONLY include the name and target percentage as of now." Not totally clear on what this is referring to - service targets?

**Review by mjac0bs — APPROVED** [review]
Thanks for the clean up!

**Review by cpathipa — APPROVED** [review]

**Pull request merged**